src/components/IOUConfirmationList.js

```diff
@@ -101,39 +102,90 @@ const defaultProps = {
+    }
+
+    /**
+     * Get unselected participants
+     * @returns {Array}
+     */
+    getUnselectedParticipants() {
+        return this.state.participants.filter(participant => !participant.selected);
+    }
+
+    /**
+     * Returns the participants with amount
+     * @param {Array} participants
+     * @returns {Array}
+     */
+    getParticipantsWithAmount(participants) {
+        return getIOUConfirmationOptionsFromParticipants(
+            participants,
+            this.props.numberFormat(this.calculateAmount(participants) / 100, {
+                style: 'currency',
+                currency: this.props.selectedCurrency.currencyCode,
+            }),
+        );
+    }
+
+    /**
+     * Returns the participants without amount
+     * @param {Array} participants
+     * @returns {Array}
+     */
+    getParticipantsWithoutAmount(participants) {
+        return participants.map(option => _.omit(option, 'descriptiveText'));
+    }
+
     /**
      * Returns the sections needed for the OptionsSelector
-     *
      * @param {Boolean} maxParticipantsReached
      * @returns {Array}
      */
     getSections() {
         const sections = [];
-
         if (this.props.hasMultipleParticipants) {
+            const selectedParticipants = this.getSelectedParticipants();
+            const unselectedParticipants = this.getUnselectedParticipants();
+
+            const formattedSelectedParticipants = this.getParticipantsWithAmount(selectedParticipants);
+            const formattedUnselectedParticipants = this.getParticipantsWithoutAmount(unselectedParticipants);
+
             const formattedMyPersonalDetails = getIOUConfirmationOptionsFromMyPersonalDetail(
                 this.props.myPersonalDetails,
-                this.props.numberFormat(this.calculateAmount() / 100, {
+                this.props.numberFormat(this.calculateAmount(selectedParticipants, true) / 100, {
                     style: 'currency',
                     currency: this.props.selectedCurrency.currencyCode,
                 }),
             );
 
-            const formattedParticipants = getIOUConfirmationOptionsFromParticipants(this.props.participants,
-                this.props.numberFormat(this.calculateAmount() / 100, {
-                    style: 'currency',
-                    currency: this.props.selectedCurrency.currencyCode,
-                }));
-
             sections.push({
                 title: this.props.translate('iOUConfirmationList.whoPaid'),
                 data: [formattedMyPersonalDetails],
                 shouldShow: true,
                 indexOffset: 0,
-            });
-            sections.push({
+            }, {
                 title: this.props.translate('iOUConfirmationList.whoWasThere'),
-                data: formattedParticipants,
+                data: [...formattedSelectedParticipants, ...formattedUnselectedParticipants],
                 shouldShow: true,
                 indexOffset: 0,
             });
```

Comment:
I'm confused on why we lost the third section that has `title: undefined`. Maybe I'm missing something, could you explain that to me?

Reply:
I removed it because  `data: [...formattedSelectedParticipants, ...formattedUnselectedParticipants]` gave me the same behaviour.

Reply:
But we don't want the unselected participants to be in the "Who Was There?" section do we?

Reply:
Hmm yeah. Seperate sections should make the code more readable.